fuse_examples/classification/knight/make_predictions_file.py

```diff
@@ -0,0 +1,111 @@
+"""
+(C) Copyright 2021 IBM Corp.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Created on June 30, 2021
+
+"""
+
+import sys
+import logging
+import os
+from typing import Optional, Union
+import pandas as pd
+
+from fuse.utils.utils_logger import fuse_logger_start
+from fuse.utils.utils_file import FuseUtilsFile
+from fuse.managers.manager_default import FuseManagerDefault
+
+from fuse_examples.classification.knight.eval.eval import TASK1_CLASS_NAMES, TASK2_CLASS_NAMES 
+
+from KNIGHT.dataset import knight_dataset
+
+def make_predictions_file(model_dir: str, 
+                          checkpoint: str, 
+                          data_path: str, 
+                          cache_path: Optional[str], 
+                          split: Union[str, dict],
+                          output_filename: str, 
+                          predictions_name: str="model.output.head_0"):
+    """
+    Automaitically make prediction files in the requested format - given path to model dir create by FuseMedML during training
+    :param model_dir: path to model dir create by FuseMedML during training
+    :param data_path: path to the original data downloaded from https://github.com/neheller/KNIGHT
+    :param cache_path: Optional - path to the cache folder. If none, it will pre-processes the data again
+    :param split: either path to pickled dictionary or the actual dictionary specifing the split between train and validation. the dictionary maps "train" to list of sample descriptors and "val" to list of sample descriptions 
+    :param output_filename: filename of the output csv file
+    :param predictions_name: the key in batch_dict of the model predictions 
+    """
+    # Logger
+    fuse_logger_start(console_verbose_level=logging.INFO)
+    lgr = logging.getLogger('Fuse')
+    lgr.info('KNIGHT: make predictions file in FuseMedML', {'attrs': ['bold', 'underline']})
+    lgr.info(f'predictions_filename={os.path.abspath(output_filename)}', {'color': 'magenta'})
+
+    # Data
+    # read train/val splits file.
+    if isinstance(split, str): 
+        split=pd.read_pickle(split)
+        if isinstance(split, list):
+            # For this example, we use split 0 out of the the available cross validation splits
+            split = split[0]
```

Comment:
same as below comment

Reply:
There is an option to provide the dictionary directly. So it's general.